src/test/kotlin/racingcar/NumberGeneratorKtTest.kt

```diff
@@ -0,0 +1,32 @@
+package racingcar
+
+import io.kotest.assertions.throwables.shouldThrowExactly
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.data.forAll
+import io.kotest.data.row
+import io.kotest.matchers.shouldBe
+
+class NumberGeneratorKtTest : StringSpec({
+    "원하는 값을 생성할 수 있다." {
+        val customNumberGenerator = CustomNumberGenerator(mutableListOf(1, 2, 3))
+        customNumberGenerator.generateWithValidation() shouldBe 1
+        customNumberGenerator.generateWithValidation() shouldBe 2
+        customNumberGenerator.generateWithValidation() shouldBe 3
+    }
+
+    "생성된 값은 0 이상 9 이하이다." {
+        forAll(
+            row(-1),
+            row(10),
+            row(11),
+            row(100),
+            row(-5),
+        ) { number ->
+            val customNumberGenerator = CustomNumberGenerator(mutableListOf(number))
```

Comment:
[Functional interface](https://kotlinlang.org/docs/fun-interfaces.html)를 활용해주신 만큼, 다음과 같이 작성해볼 수도 있을 것 같아요. (Kotlin 달달하시죠? 😆 )

```suggestion
            val numberGenerator = NumberGenerator { number }
```

Reply:
아니 이게 뭐죠..  알아서 조건에 맞는 구현체 캐스팅도 해주는건가요ㄷ